gloo/transport/tcp/context.cc

```diff
@@ -105,13 +141,13 @@ int Context::recvFromAnyFindRank(
   return -1;
 }
 
-UnboundBuffer* Context::recvFromAnyCallback(
-    int rank,
+// Allowed to be called only by ContextMutator::findRecvFromAny.
+// The context lock is already held by the caller.
```

Comment:
If I understood it correctly, it requires the lock has been acquired before calling `Pair::read()`, which is true for current use cases. Do we need to add `try_lock` checks for verification in case new calling path reaches this method? Or is it too costly to be enforced here?

Reply:
The lock will have been acquired by the ContextMutator instance that will call this function.

Reply:
I think enforcing this is costly though... It's a protected function, so only friend classes can call it, which is good enough I think.